source/NVDAObjects/UIA/winConsoleUIA.py

```diff
@@ -153,15 +153,42 @@ def expand(self, unit):
 		else:
 			return super(consoleUIATextInfo, self).expand(unit)
 
-	def _get_isCollapsed(self):
+	def compareEndPoints(self, other, which):
 		"""Works around a UIA bug on Windows 10 1803 and later."""
 		# Even when a console textRange's start and end have been moved to the
 		# same position, the console incorrectly reports the end as being
 		# past the start.
-		# Therefore to decide if the textRange is collapsed,
-		# Check if it has no text.
+		# Compare to the start (not the end) when collapsed.
+		selfEndPoint, otherEndPoint = which.split("To")
+		if selfEndPoint == "end" and not self.hasNoText:
+			selfEndPoint = "start"
+		if otherEndPoint == "End" and not other.hasNoText:
+			otherEndPoint = "Start"
+		which = f"{selfEndPoint}To{otherEndPoint}"
+		return super().compareEndPoints(other, which=which)
+
+	def setEndPoint(self, other, which):
+		"""Works around a UIA bug on Windows 10 1803 and later."""
+		# Even when a console textRange's start and end have been moved to the
+		# same position, the console incorrectly reports the end as being
+		# past the start.
+		selfEndPoint, otherEndPoint = which.split("To")
+		# In this case, there is no need to check selfEndPoint
+		# since it is about to be overwritten in the super call.
+		if otherEndPoint == "End" and not other.hasNoText:
+			otherEndPoint = "Start"
+		which = f"{selfEndPoint}To{otherEndPoint}"
+		return super().setEndPoint(other, which=which)
+
+	def _get_hasNoText(self):
```

Comment:
Since this is intented as an "internal only implementation detail for isCollapsed" I would prefer this appear to be a "private" attribute of the same name (eg `_isCollapsed`)
```suggestion
	def _get__isCollapsed(self):
```
Do we actually benefit from caching in this case? If not, I think I would prefer not to use the auto-property form and just name it `def _isCollapsed(self)`